src/tribler/core/components/libtorrent/restapi/torrentinfo_endpoint.py

```diff
@@ -106,15 +108,17 @@ async def get_torrent_info(self, request):
                 return RESTResponse({"error": str(e)}, status=HTTP_INTERNAL_SERVER_ERROR)
 
             if response.startswith(b'magnet'):
-                _, infohash, _ = parse_magnetlink(response)
-                if infohash:
-                    metainfo = await self.download_manager.get_metainfo(infohash, timeout=60, hops=hops, url=response)
+                infohash = self.get_infohash_from_magnet(response)
```

Comment:
You're not downloading the metainfo anymore.
I'm not sure this is your intention, but if it is, why even parse the magnet link? Since the metainfo will never be set, it's just going to return an error later on.

Reply:
Thank you! No, indeed, that wasn't my intention; it was a misprint during the refactoring. I've fixed it.

However, a side note: if I was able to make this change without any failed tests, it indicates suboptimal test coverage for the `TorrentInfoEndpoint`.